pymodm/fields.py

```diff
@@ -1180,15 +1199,23 @@ def contribute_to_class(self, cls, name):
                                                     self._on_delete)
 
     def to_python(self, value):
+        doc = value
         if isinstance(value, dict):
             # Try to convert the value into our document type.
-            return self.related_model.from_document(value)
-        elif isinstance(value, self.related_model):
-            return value
+            try:
+                doc = self.related_model.from_document(value)
```

Comment:
I see. For example, `value` could be a dict that represents an embedded document, which is also the primary key. That makes sense now. Thank you for your explanation. I prefer it this way to changing `from_document`.